addons/account/models/account_move_line.py

```diff
@@ -1095,6 +1095,23 @@ def action_register_payment(self, ctx=None):
             'type': 'ir.actions.act_window',
         }
 
+    def _validate_tax_alignment_with_fpos(self):
+        for line in self:
+            if not line.product_id or not line.tax_ids:
+                continue
+            company_id = self.move_id.company_id
+            mapped_tax_ids = self.env["account.fiscal.position.tax"].with_company(company_id).search([
```

Comment:
Why searching in a loop? If it can be done with `self.tax_ids`?

Reply:
Here, `self.env["account.fiscal.position.tax"]` fetch the tax mappings. Which used to align taxes with the corresponding `fiscal_position`. I don't think this will done with `self.tax_ids`